### DISCUSSION TITLE
Synchronization when using multiple uvicorn workers in fast api app

### DISCUSSION THREAD

**Discussion by kfynn007** [fastapi/fastapi#9554] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
uvicorn main:app --workers 4
```

### Description

If you have some sort web service that has multiple functionalities and one of which allows booking a limited resource. But you want to using multiple uvicorn workers to enable smooth performance, how can you ensure that there are no duplicate bookings/booking the same slot since they can happen in theory at the same time

### Operating System

macOS

### FastAPI Version

0.87.0

### Python Version

3.9.6

**Comment by jgould22** [comment, marked as answer]
There are many ways.

Usually things like this are handled in the database using things like transactions (and other functionalities, sometimes it depends on the database in use) .  If a database can offer these guarantees it is usually said to be [ACID compliant](https://en.wikipedia.org/wiki/ACID). 

Each worker will attempt to make a booking in the database, but only one will succeed while the others will receive an error.

There are many other algorithms for distributed locking but they can be considerably more complicated you wont need them for your particular example.
